From 9ede984f2b9695a06aa39a9de3d7edb92787d9bd Mon Sep 17 00:00:00 2001
From: WANG Xuerui <xen0n@gentoo.org>
Date: Wed, 19 Feb 2025 16:32:41 +0000
Subject: [PATCH 2/9] Bug 1948207 - Update gn_processor.py to support linux on
 loong64. r=firefox-build-system-reviewers,glandium

Differential Revision: https://phabricator.services.mozilla.com/D238169

(cherry picked from commit 7e63d1302021f3341923d29d151634568c0d5c2d)
---
 python/mozbuild/mozbuild/gn_processor.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/python/mozbuild/mozbuild/gn_processor.py b/python/mozbuild/mozbuild/gn_processor.py
index db7f780a1a17e..3567e1d33a7ad 100644
--- a/python/mozbuild/mozbuild/gn_processor.py
+++ b/python/mozbuild/mozbuild/gn_processor.py
@@ -183,6 +183,7 @@ def filter_gn_config(path, gn_result, sandbox_vars, input_vars, gn_target):
         "x64": "x86_64",
         "mipsel": "mips32",
         "mips64el": "mips64",
+        "loong64": "loongarch64",
     }
     oses = {
         "android": "Android",
@@ -781,7 +782,7 @@ def main():
             if target_os in ("linux", "openbsd"):
                 target_cpus.append("riscv64")
             if target_os == "linux":
-                target_cpus.extend(["ppc64", "mipsel", "mips64el"])
+                target_cpus.extend(["loong64", "ppc64", "mipsel", "mips64el"])
             for target_cpu in target_cpus:
                 vars = {
                     "host_cpu": "x64",
-- 
2.48.1

